src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const item of actions) {
+    const stateCopy = { ...currentState };
+
+    if (item.type === 'addProperties') {
+      Object.assign(stateCopy, { ...item.extraData });
+    }
+
+    if (item.type === 'removeProperties') {
+      for (const el of item.keysToRemove) {
+        delete stateCopy[el];
+      }
+    }
+
+    if (item.type === 'clear') {
+      for (const param in stateCopy) {
+        delete stateCopy[param];
+      }
+    }
```

Comment:
A default case should be added to the switch statement to handle any unexpected action types. This will ensure that there's error handling for cases where the action type does not match any of the expected types.

Reply:
ok